spec/lib/open_food_network/products_renderer_spec.rb

```diff
@@ -6,7 +6,8 @@ module OpenFoodNetwork
     let(:distributor) { create(:distributor_enterprise) }
     let(:order_cycle) { create(:simple_order_cycle, distributors: [distributor]) }
     let(:exchange) { order_cycle.exchanges.to_enterprises(distributor).outgoing.first }
-    let(:pr) { ProductsRenderer.new(distributor, order_cycle) }
+    let(:customer) { create(:customer) }
```

Comment:
I wonder if this is enough testing for such a core component.

Reply:
I'm currently writing test for the tag rule filtering part, which is scoped by customer. The ProductRenderer itself isn't really affected by `customer`.

Reply:
Okay, unit tests added in the service spec, and integration tests added in controller spec :+1: